dGame/dBehaviors/BasicAttackBehavior.cpp

```diff
@@ -178,11 +189,15 @@ void BasicAttackBehavior::DoBehaviorCalculation(BehaviorContext* context, RakNet
 		return;
 	}
 
-	const bool isImmune = destroyableComponent->IsImmune();
+	const float ImmunityTime = Game::zoneManager->GetWorldConfig()->globalImmunityTime; //DETERMINE THE DAMAGE COOLDOWN TIME
+	LOG_DEBUG("Damage cooldown time is  %f s",destroyableComponent->GetDamageCooldownTimer()); //ALSO PRINT CURRENT TIMER VALUE
+
+	const bool isImmune = (destroyableComponent->IsImmune()) || (destroyableComponent->GetDamageCooldownTimer() > 0.0f);
```

Comment:
`GetDamageCooldownTimer() > 0.0f`could be abstracted into a IsDamageImmune possibly?  not sure if a method with a similar name exists already

Reply:
Not quite sure what you mean here.

Reply:
i.e. making a public method in DestroyableComponent which is called IsDamageImmune() which does this comparison for you

Reply:
Done!